app/main.py

```diff
@@ -1,214 +1,52 @@
-
-        if "protection" in lancelot["potion"]["effect"]:
-            lancelot["protection"] += lancelot["potion"]["effect"]["protection"]
-
-        if "hp" in lancelot["potion"]["effect"]:
-            lancelot["hp"] += lancelot["potion"]["effect"]["hp"]
-
-    # arthur
-    arthur = knightsConfig["arthur"]
-
-    # apply armour
-    arthur["protection"] = 0
-    for a in arthur["armour"]:
-        arthur["protection"] += a["protection"]
-
-    # apply weapon
-    arthur["power"] += arthur["weapon"]["power"]
-
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
-
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
-
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
-
-    # mordred
-    mordred = knightsConfig["mordred"]
-
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
-
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
+from app.Ammunition.armours import Armour
+from app.Heroes.knights import Knight
+from app.Ammunition.potions import Potion
+from app.Ammunition.weapons import Weapon
+
+
+def knights_creation_config(config: dict, knight: str) -> Knight:
+    name = config[knight]["name"]
+    power = config[knight]["power"]
+    health = config[knight]["hp"]
+    weapon = Weapon(config[knight]["weapon"]["name"],
+                    config[knight]["weapon"]["power"])
+    armour = []
+    for _armour in config[knight]["armour"]:
+        armour.append(Armour(_armour["part"], _armour["protection"]))
+    potion = None
+    if config[knight]["potion"] is not None:
+        effect = config[knight]["potion"]["effect"]
+        potion = Potion(config[knight]["potion"]["name"],
+                        effect["power"] if "power" in effect else 0,
+                        effect["hp"] if "hp" in effect else 0,
+                        effect["protection"] if "protection" in effect else 0)
+    return Knight(name, power, health, weapon, armour, potion)
+
+
+def health_update(knight: Knight) -> None:
+    if knight.health < 0:
+        knight.health = 0
+
+
+def royal_battle(fighter_1: Knight, fighter_2: Knight) -> None:
+    fighter_1.health -= fighter_2.power - fighter_1.protection
+    fighter_2.health -= fighter_1.power - fighter_2.protection
+    health_update(fighter_1)
+    health_update(fighter_2)
+
+
+def battle(knights_config: dict) -> dict:
+    lancelot = knights_creation_config(knights_config, "lancelot")
+    arthur = knights_creation_config(knights_config, "arthur")
+    mordred = knights_creation_config(knights_config, "mordred")
+    red_knight = knights_creation_config(knights_config, "red_knight")
```

Comment:
You can iterate through all knights in dict and create Knight instances 
```suggestion
    knights = {}
    for knight in knights_config:
          knights[knight['name']] = create_knight(knight)
```